src/standalone/Soil/rre.jl

```diff
@@ -330,84 +347,76 @@ end
 
 
 """
-    RichardsTridiagonalW{R, J, W, T} <: ClimaLand.AbstractTridiagonalW
+    ImplicitEquationJacobian{M, S}
```

Comment:
https://github.com/CliMA/ClimaLand.jl/blob/ec37455a0e58d6662c44a7216a84cc16042fa8f5/src/standalone/Soil/rre.jl#L364

I see this is defined only in rre.jl. Should this be defined in Soil.jl (the same struct will also be used for the full soil model)? Or, if not, should this be defined something more specific (RichardsImplicitEquationJacobian)?

Reply:
I think the same struct will be used for the full soil model, since they'll both have tridiagonal jacobians, and this struct is generalized for any soil model rather than just richards